ray-operator/controllers/ray/common/job.go

```diff
@@ -54,96 +54,136 @@ func GetMetadataJson(metadata map[string]string, rayVersion string) (string, err
 	return pkgutils.ConvertByteSliceToString(metadataBytes), nil
 }
 
-// GetK8sJobCommand builds the K8s job command for the Ray job.
-func GetK8sJobCommand(rayJobInstance *rayv1.RayJob) ([]string, error) {
-	address := rayJobInstance.Status.DashboardURL
+// BuildJobSubmitCommand builds the job submit command based on submission mode.
+func BuildJobSubmitCommand(rayJobInstance *rayv1.RayJob, submissionMode rayv1.JobSubmissionMode) ([]string, error) {
+	var address string
+	var waitLoop []string
+
+	switch submissionMode {
+	case rayv1.SidecarMode:
+		// Sidecar submitter talks to the local head Pod dashboard.
+		address = "http://127.0.0.1:8265"
```

Comment:
Can you add a TODO and open an issue to track the progress to make sure it works when users use different port for Ray dashboard?

Reply:
Maybe add a function to get Ray dashboard port from the RayJob CR.